gpbft/gpbft.go

```diff
@@ -1086,6 +1087,24 @@ func (q *quorumState) HasStrongQuorumFor(key ChainKey) bool {
 	return ok && supportForChain.hasStrongQuorum
 }
 
+// mayHaveStrongQuorumFor checks whether the given chain can possibly reach
+// strong quorum.
+func (q *quorumState) mayHaveStrongQuorumFor(key ChainKey) bool {
+	supportForChain, found := q.chainSupport[key]
+	if !found {
+		// There is no support for given chain. Strong quorum is only possible if the
+		// aggregate power of voted participants dos not exceed ⅔ of total power.
+		return !q.ReceivedFromStrongQuorum()
+	}
+
+	// A strong quorum is only feasible when the total support for the given chain,
+	// combined with the aggregate power of not yet voted participants, exceeds ⅔ of
+	// total power.
+	nonSendersTotalPower := NewStoragePower(0).Sub(q.powerTable.Total, q.sendersTotalPower)
+	strongQuorumThreshold := NewStoragePower(0).Add(nonSendersTotalPower, supportForChain.power)
```

Comment:
I don't think threshold is the right word here. Maybe `possibleSupportForKey`, or `maximumSupportForKey`?